src/ds3.c

```diff
@@ -483,7 +483,13 @@ static ds3_error* _net_process_request(const ds3_client* client, const ds3_reque
             error->error = g_new0(ds3_error_response, 1);
             error->error->status_code = response_data.status_code;
             error->error->status_message = ds3_str_init(response_data.status_message->value);
-            error->error->error_body = ds3_str_init((char*)response_data.body->data);
+
+            if (response_data.body != NULL && response_data.body->data != NULL) {
+                error->error->error_body = ds3_str_init((char*)response_data.body->data);
+            }
+            else {
+                error->error->error_body = ds3_str_init("(empty)");
+            }
```

Comment:
This looks old.  This method was moved to ds3_net.c - are you still seeing an issue with it?

It has been changed to:

34 <   if (response_data.status_code < 200 || response_data.status_code >= 300) {
535 <       ds3_error* error = ds3_create_error(DS3_ERROR_BAD_STATUS_CODE, "Got an unexpected status code.");
536 <       error->error = g_new0(ds3_error_response, 1);
537 <       error->error->http_error_code = response_data.status_code;
538 <       error->error->code = ds3_str_init(response_data.status_message->value);
539 <       if (response_data.body != NULL) {
540 <           error->error->message = ds3_str_init_with_size((char*)response_data.body->data, response_data.bod>
541 <           g_byte_array_free(response_data.body, TRUE);
542 <       } else {
543 <           ds3_log_message(client->log, DS3_ERROR, "The response body for the error is empty");
544 <           error->error->message = NULL;
545 <       }
546 <       ds3_string_multimap_free(response_headers);
547 <       ds3_str_free(response_data.status_message);
548 <       g_free(url);